internal/fields.go

```diff
@@ -0,0 +1,16 @@
+package internal
+
+import "sort"
+
+// SortedKeys returns a sorted list of non-empty field keys
+func SortedKeys(fields map[string]any) []string {
```

Comment:
```go
func SortedKeys(fields map[string]any) []string {
    keys := make([]string, 0,len(fields))
    delete(fields, "")
    for key := range fields {
        keys = append(keys,key)
    }
    sort.Strings(keys)
    return keys
}
```

Reply:
are you sure the delete is cheaper than the check?

Reply:
Probably the same for normal sized maps, and I don't think it will have to manage ones with many million keys. GTG